datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -273,6 +275,7 @@ impl MemoryConsumer for ExternalSorter {
             &self.expr,
             self.session_config.batch_size(),
             tracking_metrics,
+            self.fetch,
```

Comment:
Nice thing is that it also reduces disk spilling, as sort + limit is done before writing.

Reply:
Yeah maybe the spilling could see the remaining batch is so small it could add the sorted data to memory again - avoiding the spill 🤔

Reply:
As a future PR / optimization perhaps